frontend/src/components/TaskMap.tsx

```diff
@@ -207,9 +225,18 @@ export const TaskMap: FC<{
       roadmapId: roadmapId!,
       relation: { from, to, type },
     });
+    saveNodeState();
     setIsLoading(false);
   };
 
+  const resetCanvas = () => {
+    localStorage.removeItem(flowKey);
+    drawCanvas();
+  };
+
+  const nodeTypes = useMemo(() => ({ special: CustomNodeComponent }), []);
+  const edgeTypes = useMemo(() => ({ custom: CustomEdge }), []);
```

Comment:
I guess it would make sense to define these outside the component as these are constant (no dependencies), and thus the `useMemo` isn't really needed for anything.

Reply:
Good point 👍